solutions/gold/usaco-1041.mdx

```diff
@@ -179,4 +194,72 @@ public class Haircut {
 ```
 
 </JavaSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+typedef long long ll;
+
+// The C++ Solution uses Segement Tree instead BIT,
+// but the idea is still the same.
+
```

Comment:
can you replace the purs implementation with a class from [here](https://usaco.guide/gold/PURS?lang=cpp)?

Reply:
the `SumSegmentTree ` one?

Reply:
ye